### PR TITLE
fix: generate correct ui5.yaml for adp

### PR DESCRIPTION
**Pull request by GDamyanov**
Fix for #2145 

- Configure custom middlewares in ui5.yaml in correct way

### CODE REVIEW

File: packages/ui5-config/src/types/middlewares.ts [hunk 1]
```diff
@@ -38,6 +38,7 @@ export interface FioriToolsProxyConfig {
     backend?: FioriToolsProxyConfigBackend[];
     ui5?: Partial<FioriToolsProxyConfigUI5>;
     ignoreCertError?: boolean;
+    afterMiddleware?: string;
```
Comment (tobiasqueck) [inline packages/ui5-config/src/types/middlewares.ts:41]: Please remove this here again. The `afterMiddleware` property is not part of the proxy config, it is a separate property for every middleware

File: packages/adp-tooling/src/writer/options.ts [hunk 2]
```diff
@@ -110,7 +110,8 @@ function addFioriToolsMiddlwares(ui5Config: UI5Config, config: AdpWriterConfig)
             url: config?.ui5?.frameworkUrl,
             version: config?.ui5?.minVersion ?? '' //default to latest if version is not set
         },
-        backend: [backendConfig as FioriToolsProxyConfigBackend]
+        backend: [backendConfig as FioriToolsProxyConfigBackend],
+        afterMiddleware: 'fiori-tools-preview'
```
Comment (tobiasqueck) [inline packages/adp-tooling/src/writer/options.ts:114]: I don't object that change but can you explain why this is a problem? I am afraid we are not understanding the root cause here.

**Review by tobiasqueck — CHANGES_REQUESTED** [review]

File: packages/ui5-config/src/types/middlewares.ts (hunk 1 above)
Comment (GDamyanov) [reply packages/ui5-config/src/types/middlewares.ts]: Fixed in [9310a2a](https://github.com/SAP/open-ux-tools/pull/2146/commits/9310a2aec0854ff120effa183bb33a63b15e221f)

**Review by tobiasqueck — APPROVED** [review]
* changeset ok
* issue correctly fixed (tested it locally)
* test snapshots correctly updated
* readme update explaining the requirement

File: packages/adp-tooling/src/writer/options.ts (hunk 2 above)
Comment (nikmace) [reply packages/adp-tooling/src/writer/options.ts]: Readme updated in: [619ae1e](https://github.com/SAP/open-ux-tools/pull/2146/commits/619ae1e4d64848291997270a4134034d24b97707)

**Pull request merged**